### DISCUSSION TITLE
API change name of note

### DISCUSSION THREAD

**Discussion by Sebiann** [TriliumNext/Trilium#2057] [Q&A] [answered]
I want to use a script to change the name of a note

**Comment by abitofevrything** [comment, marked as answer]
You can use this:
```js
async function setNoteTitle(noteId, title) {
    return await api.runOnBackend((noteId, title) => {
        const becca = require('../becca/becca');
        const noteService = require('./notes');
        
        const note = becca.getNote(noteId);

        if (!note) {
            return [404, `Note ${noteId} has not been found`];
        }

        if (!note.isContentAvailable()) {
            return [400, `Note ${noteId} is not available for change`];
        }

        const noteTitleChanged = note.title !== title;

        note.title = title;

        note.save();

        if (noteTitleChanged) {
            noteService.triggerNoteTitleChanged(note);
        }
    }, [noteId, title]);
}
```

This is (nearly) the exact same code used normally for renaming notes, which you can find [here](https://github.com/zadam/trilium/blob/master/src/routes/api/notes.js#L190).

**Reply by zadam** [reply]
Note that this works only in master, not in the released version (there was a big change in backend/frontend entity handling). This also covers quite a lot of corner cases, IMHO just for scripting use case I would use something simpler (assuming backend script note):

```
const note = api.getNote(noteId);
note.title = 'New title';
note.save();
```

(this should work in both stable and master branches).

**Reply by Sebiann** [reply]
thx. I used the simpler version Zadam recommended.

**Reply by Sebiann** [reply]
@zadam 
```js
const note = api.originEntity.getNote();

const date = note.getLabelValue('date');
const arc = note.getLabelValue('arc');
const part = note.getLabelValue('part');

note.title = `${date} - ${arc} - Part ${part}`;
note.save();
```
so ive been testing with this code. and running on attribute change. but for some reason date and part always work. but arc only works sometimes.
it doesnt even show null in the name
